qunit/qunit-test.ts

```diff
@@ -0,0 +1,556 @@
+  assert.ok( NaN, "NaN fails" );
+  assert.ok( "", "empty string fails" );
+  assert.ok( null, "null fails" );
+  assert.ok( undefined, "undefined fails" );
+});
+
+QUnit.test( "a test", function( assert ) {
+  assert.equal( 1, "1", "String '1' and number 1 have the same value" );
+});
+
+
+QUnit.test( "equal test", function( assert ) {
+  assert.equal( 0, 0, "Zero, Zero; equal succeeds" );
+  assert.equal( "", 0, "Empty, Zero; equal succeeds" );
+  assert.equal( "", "", "Empty, Empty; equal succeeds" );
+ 
+  assert.equal( "three", 3, "Three, 3; equal fails" );
+  assert.equal( null, false, "null, false; equal fails" );
+});
+
+QUnit.test( "a test", function( assert ) {
+  assert.expect( 2 );
+ 
+  function calc( x: number, operation: (x:number)=> number ) {
+    return operation( x );
+  }
+ 
+  var result = calc( 2, function( x ) {
+    assert.ok( true, "calc() calls operation function" );
+    return x * x;
+  });
+ 
+  assert.equal( result, 4, "2 squared equals 4" );
+});
+
+
+QUnit.test( "notDeepEqual test", function( assert ) {
+  var obj = { foo: "bar" };
+ 
+  assert.notDeepEqual( obj, { foo: "bla" }, "Different object, same key, different value, not equal" );
+});
+
+
+QUnit.test( "a test", function( assert ) {
+  assert.notEqual( 1, "2", "String '2' and number 1 don't have the same value" );
+});
+
+QUnit.test( "notOk test", function( assert ) {
+  assert.notOk( false, "false succeeds" );
+  assert.notOk( "", "empty string succeeds" );
+  assert.notOk( NaN, "NaN succeeds" );
+  assert.notOk( null, "null succeeds" );
+  assert.notOk( undefined, "undefined succeeds" );
+ 
+  assert.notOk( true, "true fails" );
+  assert.notOk( 1, "1 fails" );
+  assert.notOk( "not-empty", "not-empty string fails" );
+});
+
+
+QUnit.test( "notPropEqual test", function( assert ) {
+  class Foo {
+    x: any;
+    y: any;
+    z: any;
+    constructor( x: any, y: any, z: any ) {
+      this.x = x;
+      this.y = y;
+      this.z = z;
+    }
+    
+    doA = function () {};
+    doB = function () {};
+    bar = 'prototype';
+  }
+  var foo = new Foo( 1, "2", [] );
+  var bar = new Foo( "1", 2, {} );
+  assert.notPropEqual( foo, bar, "Properties values are strictly compared." );
+});
+
+
+QUnit.test( "a test", function( assert ) {
+  assert.notStrictEqual( 1, "1", "String '1' and number 1 have the same value but not the same type" );
+});
+
+QUnit.test( "propEqual test", function( assert ) {
+  class Foo {
+    x: any;
+    y: any;
+    z: any;
+    constructor( x: any, y: any, z: any ) {
+      this.x = x;
+      this.y = y;
+      this.z = z;
+    }
+    
+    doA = function () {};
+    doB = function () {};
+    bar = 'prototype';
+  }
+ 
+  var foo = new Foo( 1, "2", [] );
+  var bar: any = {
+    x : 1,
+    y : "2",
+    z : []
+  };
+  assert.propEqual( foo, bar, "Strictly the same properties without comparing objects constructors." );
+});
+
+// QUnit.assert['mod2'] = function( value: any, expected: any, message: string ): any {
```

Comment:
Is there a recommend way to test this? I've left the example commented out because of the no implicit any option.